src/rules/no-nested-type-without-ref.ts

```diff
@@ -0,0 +1,35 @@
+import {
+    Rule,
+    ValidationError,
+    ApiSpec,
+    getAllObjectProperties,
+    hasProperty,
+    getProperty
+} from '../Rule';
+
+export default class NoNestedTypeWithoutRef implements Rule {
+    describe() {
+        return 'When nesting types, the nested type must either be a primitive type or a reference type';
+    }
+
+    validate(json: ApiSpec) {
+        getAllObjectProperties(json, 'components.schemas').forEach(({ object }) => {
+            if (hasProperty(object, 'properties')) {
+               getAllObjectProperties(object, 'properties').forEach(({path, name, object}) => {
+                    if (hasProperty(object, 'type')) {
+                        const type = getProperty(object, 'type');
+                        if (type === 'array' && hasProperty(object, 'items')) {
+                            const prop = Object.keys(getProperty(object, 'items') as any)[0];
```

Comment:
Looks like it only checks the first property?

`getProeprty(object, 'items.type')` ?

Reply:
`getProperty(object, 'items.type')` doesn't work, but I updated it in another way and it checks not only the first property now.